### PR TITLE
Backport 0.57.9 reverts

### PR DESCRIPTION
**Pull request by umanwizard**
This is a backport of the changes in https://github.com/MaterializeInc/materialize/pull/20116 onto main. We have strong evidence that it is necessary for fixing a large memory usage regression at least for the hierarchical min/max reduce, and possibly for other operators as well.

`mz_dataflow_shutdown_durations_histogram` gets reverted as collateral damage, as it depended on refactors done for the arrangement size logging work, and so it was very difficult to cleanly revert one without the other. @teskje (the author of `mz_dataflow_shutdown_durations_histogram`) has confirmed that it's okay to revert both of these for now.

Fixes https://github.com/MaterializeInc/materialize/issues/20125

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Temporarily remove the size, capacity, and allocations fields from `mz_arrangement_sizes` and `mz_dataflow_arrangement_sizes`, pending investigation of an underlying bug.
  - Temporarily remove `mz_dataflow_shutdown_durations_histogram`, which was necessary in order to make the aforementioned revert easier.

### CODE REVIEW

**Review by benesch — APPROVED** [review]
LGTM to unblock the release, but @teskje should take a look post merge too.

**Comment by umanwizard** [comment]
Agreed, this is complicated enough that I don't want it to go to prod without Jan checking. But it's good enough to get the release cycle unblocked.

File: src/timely-util/src/operator.rs
Comment (teskje) [inline src/timely-util/src/operator.rs:574]: This accidentally reverts @vmarcos' fix hashing fix from https://github.com/MaterializeInc/materialize/pull/19848.

Not critical as the bug this fixes only occurs when the monotonic one-off SELECTs feature is enabled, which is not the case in prod currently. However, we should make sure to restore the fix ASAP.

**Review by teskje — APPROVED** [review]
LGTM, apart from the lost ahash fix mentioned.

I wonder though... https://github.com/MaterializeInc/materialize/pull/19848/files contains a regression test that should have caught the revert of the fix. Is that just not part of the CI suite or is something wrong with the test? (cc @vmarcos)

**Pull request merged**
